File: fathersDay/Dsa/Day 35 ( 191. Number of 1 Bits)/Solution.py
class Solution:
    def NumberOfoneBits(self, n: int) -> int:
        count = 0
        while n:
            count += n & 1
            n >>= 1
        return count
# Example usage:
solution = Solution()
n = 11  # Binary representation: 1011
result = solution.NumberOfoneBits(n)
print(result)  # Output: 3 (there are three '1' bits in the

File: fathersDay/Dsa/Day 35 ( 191. Number of 1 Bits)/Solution.cpp
#include <iostream>

class Solution {
public:
    int NumberOfoneBits(int n) {
        int count = 0;
        while (n) {
            count += n & 1;
            n >>= 1;
        }
        return count;
    }
};

int main() {
    Solution solution;
    int n = 11; // Binary representation: 1011
    int result = solution.NumberOfoneBits(n);
    std::cout << result << std::endl; // Output: 3
    return 0;
}
